deployments/apiv2/services/users/src/main.py

```diff
@@ -982,6 +1013,83 @@ async def get_user(
         raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
 
+@app.put("/customers/{account_id}", status_code=status.HTTP_204_NO_CONTENT)
+async def update_customer(
+    request: Request,
+    details: AccountUpdateAction,
+    account_id: uuid.UUID,
+    token=Depends(ProtectedAny(scopes=[Scopes.CURI__ADMIN])),
+):
+    """Update a customer account's information in the database, restricted use for Curi Admin.
+
+    The action to take on the user should be passed in the body of PUT request as action_type:
+        - deactivate: set suspended field to true
+        - reactivate: set sespended field to false and set login attempts to 0
+        - edit: set usage restrictions in customers and update account scopes for customer, remove from users if present
+    """
+    self_id = uuid.UUID(hex=token.account_id)
+    action = details.action_type
+    update_query = None
+
+    try:
+        bind_context_to_logger(
+            {
+                "user_id": None,
+                "customer_id": str(self_id),
+                "action": action,
+                "target_customer": str(account_id),
+            }
+        )
+
+        if action == "deactivate":
+            update_query = "UPDATE customers SET suspended='t' WHERE id=$1"
+            query_args = (account_id,)
+        elif action == "reactivate":
+            # when reactivated, failed login attempts should be set back to 0.
+            update_query = "UPDATE customers SET suspended='f', failed_login_attempts=0 WHERE id=$1"
+            query_args = (account_id,)
+        elif action == "edit":
+            update_query = "UPDATE customers SET usage_restrictions=$1 WHERE id=$2"
+            query_args = (json.dumps(details.usage), account_id)
+
+            async with request.state.pgpool.acquire() as con:
+                async with con.transaction():
+                    # get current customer scopes
+                    rows = await con.fetch(
+                        "SELECT scope FROM account_scopes WHERE customer_id=$1 AND user_id IS NULL",
+                        account_id,
+                    )
+
+                    # get product name from admin scopes
+                    existing_products = [dict(p)["scope"].split(":")[0] for p in rows]
```

Comment:
We have a function `get_product_tags_of_admin` that handles this

Reply:
But the returned value from the fetch query would still need to be iterated through and the scopes in the db are strings so `s.tags` fails in `get_product_tags_of_admin`. Additionally, it needs to be compared against the list of strings sent in the request body

Reply:
I think something like `get_product_tags_of_admin([convert_scope_str(row["scope"]) for row in rows])` should work. `Scopes` inherits from `StrEnum` so all the fields have all the string behavior. We also have `ScopeConverter` which our models can inherit from which also converts the strings to `Scopes`

Reply:
This current method assumes that the first component of each scope is a product name which could change if we add a scope in the future which does not start with a product name